File: backend/src/test/java/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckCodeExtractorTest.java
package info.gamewise.lor.videos.deckcodeextractor;

import org.junit.jupiter.api.Test;

import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertEquals;

class DescriptionWithDeckCodeExtractorTest {

    private final DeckCodeExtractor extractor = new DescriptionWithDeckCodeExtractor();

    @Test
    void deckCodeAndUrl() {
        final var deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA";

        final var description = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/ANY-DECK-CODE
            
            Deck code: %s
            
            Thank you!
            Have a nice day.
        """.formatted(deckCode);

        Optional<String> deckCodeOpt = extractor.extract(description);

        assertTrue(deckCodeOpt.isPresent());
        assertEquals(deckCodeOpt.get(), deckCode);
    }

    @Test
    void onlyUrlShouldBeEmpty() {
        final var description = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/ANY-DECK-CODE
            
            Thank you!
            Have a nice day.
        """;

        Optional<String> deckCodeOpt = extractor.extract(description);

        assertTrue(deckCodeOpt.isEmpty());
    }

}

File: backend/src/test/kotlin/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckCodeExtractorTest.kt
package info.gamewise.lor.videos.deckcodeextractor

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

internal class DescriptionWithDeckCodeExtractorTest {

    private val extractor: DeckCodeExtractor = DescriptionWithDeckCodeExtractor()

    @Test
    fun deckCodeAndUrl() {
        val deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA"
        val description: String = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/ANY-DECK-CODE
            
            Deck code: $deckCode
            
            Thank you!
            Have a nice day.
        """.trimIndent()

        val deckCodeOpt: String? = extractor.extract(description)
        assertNotNull(deckCodeOpt)
        assertEquals(deckCodeOpt, deckCode)
    }

    @Test
    fun onlyUrlShouldBeEmpty() {
        val description: String = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/ANY-DECK-CODE
            
            Thank you!
            Have a nice day.
        """.trimIndent()

        val deckCode: String? = extractor.extract(description)
        Assertions.assertNull(deckCode)
    }
}